### PR TITLE
Add filters to abuse reports admin page

### LINKED ISSUE #20417: Add category filter to abuse reports page
## Is this feature related to a problem? Please describe.

As a support admin, I want to be able to see the critical support queries I should address first, so I can be efficient with my support tasks.

## Describe the solution you’d like

- Add category filter to abuse report page
- Show the category on the collapsed view
- Collapse reports by default

### PR DESCRIPTION
**Pull request by PhilipHow**
## What type of PR is this? (check all applicable)

- [ ] Refactor
- [x] Feature
- [ ] Bug Fix
- [ ] Optimization
- [ ] Documentation Update

## Description

https://www.loom.com/share/429d6117d7734cd3805679224294d44d

## Related Tickets & Documents

- Close #20417

## QA Instructions, Screenshots, Recordings

[_Please replace this line with instructions on how to test your changes, a note
on the devices and browsers this has been tested on, as well as any relevant
images for UI changes._](https://www.loom.com/share/429d6117d7734cd3805679224294d44d)

### UI accessibility checklist
_If your PR includes UI changes, please utilize this checklist:_
- [x] Semantic HTML implemented?
- [] Keyboard operability supported?
- [] Checked with [axe DevTools](https://www.deque.com/axe/) and addressed `Critical` and `Serious` issues?
- [] Color contrast tested?

## Added/updated tests?
_We encourage you to keep the code coverage percentage at 80% and above._

- [ ] Yes
- [x ] No, and this is why: because I'm a coward and this is a small change
- [ ] I need help with writing tests

### CODE REVIEW

File: app/views/admin/feedback_messages/_feedback_message.html.erb [hunk 1]
```diff
@@ -4,15 +4,17 @@
   <% feedback_message_emails = @email_messages.select { |email| email.feedback_message_id == feedback_message.id } %>
 <% end %>
 
-<details open class="crayons-card" id="accordion-<%= feedback_message.id %>">
```
Comment (maestromac) [inline app/views/admin/feedback_messages/_feedback_message.html.erb:7]: Was this an intentional change? is this not something we want open by default?

File: app/views/admin/feedback_messages/_feedback_message.html.erb [hunk 2]
```diff
@@ -4,15 +4,17 @@
...
+<details class="crayons-card" id="accordion-<%= feedback_message.id %>">
   <summary class="p-5">
     <span id="header<%= feedback_message.id %>">
       <span id="collapse__header__link-<%= feedback_message.id %>">
         Report #<%= feedback_message.id %> - Submitted: <%= time_ago_in_words feedback_message.created_at %> ago
       </span>
       <div class="float-right">
+       <span class="c-indicator c-indicator--warning crayons-subtitle-2"><%= feedback_message.category.titleize %></span>
```
Comment (maestromac) [inline app/views/admin/feedback_messages/_feedback_message.html.erb:14]: Nice touch!

File: app/views/admin/feedback_messages/index.html.erb [hunk 3]
```diff
@@ -11,6 +11,9 @@
       <%= f.label :reporter_username_cont, "Reporter", class: "crayons-field__label m-1 screen-reader-only" %>
       <%= f.search_field :reporter_username_cont, placeholder: "Reporter", class: "crayons-textfield w-auto m-1" %>
       <%= f.select(:status_eq, options_for_select(%w[Open Invalid Resolved], params[:status] || "Open"), { include_blank: true }, class: "crayons-select w-auto m-1") %>
+     <%= f.label :category_eq, "Category", class: "crayons-field__label m-1 screen-reader-only" %>
+<%= f.select :category_eq, options_for_select(FeedbackMessage::CATEGORIES, params[:category]), { include_blank: true }, class: "crayons-select w-auto m-1" %>
```
Comment (maestromac) [inline app/views/admin/feedback_messages/index.html.erb:15]: nit: I would peronsally end this bar with the status (open/close) but I'll leave that up to others.

**Review by maestromac — APPROVED** [review]
LGTM!

**Review by benhalpern — APPROVED** [review]

File: app/controllers/admin/feedback_messages_controller.rb [hunk 4]
```diff
@@ -121,10 +121,15 @@ def feedback_message_params
...
+      params[:q] ||= {}
+      if params.dig(:q, :status_eq).blank? && params[:status].present?
         params[:q][:status_eq] = params[:status]
       end
-      params[:status] = params[:q][:status_eq] if params[:q].present? # ransack
+      if params.dig(:q, :category_eq).blank? && params[:category].present?
+        params[:q][:category_eq] = params[:category]
+      end
+      params[:status] = params[:q][:status_eq] if params[:q].present?
+      params[:category] = params[:q][:category_eq] if params[:q].present?
```
Comment (maestromac) [inline app/controllers/admin/feedback_messages_controller.rb:132]: this section a bit of a doozy though. should refactor afterward

**Pull request merged**